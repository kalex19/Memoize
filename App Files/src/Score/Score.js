import React, { Component } from 'react';
import Style from './Score.scss';

export default class Score extends Component {
  constructor(props){
  super(props);
}

render () {
    return (
            <div className={this.props.start ? "App-hidden" : "Score"}>
                <h1 className="Score-text">SCORE:</h1>
                <p className={this.props.scoreAnimation ? "Score-animation" : "Score-text"}>{this.props.score} pts</p>
            </div>
        )
    }
}