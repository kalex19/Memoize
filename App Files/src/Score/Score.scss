@import '../variables.scss';

.Score {
    display: block;
    justify-content: center;
    justify-content: space-between;
    height: 15vh;
    width: 25%;
    margin: 0 2%;
    border: dashed 1px $main-color;
  }

  .Score-animation {
      font-size: 1em;
      color: yellow;
      animation: bounceIn 2s;
      pointer-events: none;
      margin: 1%;
    }
    
  .Score-text {
    font-size: .6em;
    color: $main-color;
  }

  .App-hidden {
    visibility: hidden;
    width: 40%;
    margin: 0;
  }

  @keyframes bounceIn {
    0% {
      transform: scale(0.1);
      opacity: 0;
    }
    60% {
      transform: scale(1.2);
      opacity: 1;
    }
    100% {
      transform: scale(1);
    }
  }
