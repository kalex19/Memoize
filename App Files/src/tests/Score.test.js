import React from 'react';
import Score from '../Score/Score.js';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Score', () => {
    let wrapper;
  
    beforeEach(() => {
      wrapper = shallow(
        <Score
        />
      );
    });

    it('should match the snapshot with all data passed in', () => {
        expect(wrapper).toMatchSnapshot();
    });

  });