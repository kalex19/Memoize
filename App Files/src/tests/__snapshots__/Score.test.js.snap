// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score should match the snapshot with all data passed in 1`] = `
<div
  className="Score"
>
  <h1
    className="Score-text"
  >
    SCORE:
  </h1>
  <p
    className="Score-text"
  >
     pts
  </p>
</div>
`;
